src/app/server/AppDelegate.h

```diff
@@ -29,11 +29,26 @@ class AppDelegate
 public:
     virtual ~AppDelegate() {}
     /**
-     * This is called on start of session establishment process
+     * This is called when the PBKDFParamRequest is received and indicates the start of the session establishment process
      */
     virtual void OnCommissioningSessionEstablishmentStarted() {}
+
+    /**
+     * This is called when the commissioning session has been established
+     */
     virtual void OnCommissioningSessionStarted() {}
-    virtual void OnCommissioningSessionStopped(CHIP_ERROR err) {}
+
+    /**
+     * This is called when there is an error in establishing a commissioning session (such as, when an invalid passcode is provided)
+     *
+     * @param err CHIP_ERROR indicating the error that occurred during session establishment
+     */
+    virtual void OnCommissioningSessionEstablishmentError(CHIP_ERROR err) {}
```

Comment:
Could this return bool for `close window`, such that the callback can decide whether the window should be closed right away?

Reply:
Done. Made it return a bool for close window. See updated diff.

Reply:
Agreed to revert this update after this conversation: https://github.com/project-chip/connectedhomeip/pull/28240#discussion_r1275398506